src/sage/rings/polynomial/polynomial_ring_constructor.py

```diff
@@ -35,6 +35,9 @@ class padic_base_leaves:
 import sage.rings.abc
 from sage.rings.integer import Integer
 from sage.rings.finite_rings.finite_field_base import FiniteField
+from sage.rings.semirings.tropical_semiring import TropicalSemiring
+from sage.rings.semirings.tropical_polynomial import TropicalPolynomialSemiring
+from sage.rings.semirings.tropical_mpolynomial import TropicalMPolynomialSemiring
```

Comment:
```suggestion
from sage.rings.semirings.tropical_semiring import TropicalSemiring
```
Do the other two imports where you actually need them.